### PR TITLE
Allow tangential boundary conditions for GMG global coarsening.

### PR DESCRIPTION
**Pull request by quangx**
This pull request add a 2D shell GMG global coarsening  test and removes an assert for tangential boundary velocity indicators.  The assert was in place because GMG global coarsening had not been tested for this case. Given that the iteration counts for the 2D shell with local smoothing and global coarsening are similar for the 2D shell (also tested with a finer mesh), I think we can safely remove the assert.

### CODE REVIEW

File: source/simulator/solver/stokes_matrix_free_global_coarsening.cc [hunk 1]
```diff
@@ -1574,7 +1574,7 @@ namespace aspect
 
             if (this->get_boundary_velocity_manager().get_tangential_boundary_velocity_indicators().size() > 0)
               {
-                Assert(false, dealii::StandardExceptions::ExcNotImplemented("This is not tested."));
+                // Assert(false, dealii::StandardExceptions::ExcNotImplemented("This is not tested."));
```
Comment (bangerth) [inline source/simulator/solver/stokes_matrix_free_global_coarsening.cc]: If you remove the assert, just remove it. The current state of the code base should represent the current state, not some previously present assertion -- that is just bound to confuse readers. If you think the code is now well tested, let's just remove the assertion. If you *don't* think that the code is well tested, then of course we should not remove the assertion.

Comment (quangx) [reply source/simulator/solver/stokes_matrix_free_global_coarsening.cc]: From a bit more testing I think we are good to remove the assert. I will do that in this pr.
